fishtank/src/cluster.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Docker } from './backend'
+
+export const DEFAULT_IMAGE = 'ironfish:latest'
+
+export class Cluster {
+  public readonly name: string
+
+  private readonly backend: Docker
+
+  constructor(options: { name: string; backend?: Docker }) {
+    this.name = options.name
+    this.backend = options.backend ?? new Docker()
+  }
+
+  private containerName(name: string): string {
```

Comment:
This function is more like getClusterName(), because it basically get the cluster name and attach with user entered node/container name. 

Since a cluster can have multiple containers, do we need to keep track of the containers with an array? This way its also easy to implement `get(nodeName: string) -> Node` in the next step.  If we do this way, then this function will be `addContainer()` and it will store the name and return the name

Reply:
This class cannot rely on any in-memory state, because the program may be terminated and resumed at any point. If we want to build a command line tool on top of this class, then you need to keep in mind that, after each invocation of the tool, anything stored in memory will be erased.

The proper thing to do is to simply rely on Docker's state, which means: not storing all the spawned containers in an array, but asking Docker which containers have been spawned.

Reply:
So it sounds ike the containerName function is more like a util/helper function for cluster class. If thats the case, I am fine with the current implementation.
Nit, it would be nice to have the public function first, then protected and then private.